site/app/libraries/database/DatabaseQueries.php

```diff
@@ -2270,7 +2270,7 @@ public function getTeamsByGradeableAndRotatingSections($g_id, $sections, $orderB
     }
 
     public function getTotalComponentCount($g_id) {
-        $this->course_db->query("SELECT count(*) AS cnt FROM gradeable_component WHERE g_id=?", [$g_id]);
+        $this->course_db->query("SELECT count(*) AS cnt FROM gradeable_component WHERE g_id=? AND gc_is_peer=?", [$g_id, $this->course_db->convertBoolean(false)]);
```

Comment:
It looks like this change makes it so that getTotalComponentCount only returns the non-peer components. Wouldn't a better fix be to include peer graded components in the %complete calculation?

Reply:
Not really, because the progress bar for the TA side should be calculated using the TA-completed components/TA-only components as stated in the expected behavior for the issue. The progress bar for the peer side should be calculated through peer-completed components/peer components, but Nia has another PR that addresses the peer progress bar, so this PR is specifically aimed at resolving the issues with the TA progress bar.

Reply:
Thanks for the feedback. I agree that the getTotalComponentCount() should only return all components. I created a new function named getTaComponentCount()